src/learner.cc

```diff
@@ -528,6 +528,9 @@ class LearnerConfiguration : public Learner {
     auto const& objective_fn = learner_parameters.at("objective");
     if (!obj_) {
       CHECK_EQ(get<String const>(objective_fn["name"]), tparam_.objective);
+      if (ctx_.IsSycl()) {
+        tparam_.objective = ObjFunction::GetSyclImplementationName(tparam_.objective);
+      }
```

Comment:
Is it possible that the dispatching happens inside `ObjFunction::Create`? Also, what happens if a user saves the model, loads it back on a different machine that doesn't have a sycl device?

Reply:
It is a good idea. In this case we wouldn't have problems with saved models.